flytekit/common/tasks/task.py

```diff
@@ -268,6 +273,59 @@ def _python_std_input_map_to_literal_map(self, inputs):
             for k, v in _six.iteritems(self.interface.inputs)
         })
 
+    def _produce_deterministic_version(self, version=None):
+        """
+        :param Text version:
+        :return Text:
+        """
+
+        if self.container is not None and self.container.data_config is None:
```

Comment:
so no white list?  can you add a comment explaining why data config can't be none?

Reply:
done. i'd rather not add a whitelist because barring the raw container task anything that's containerless (e.g. Sagemaker) in the future should just seamlessly be able to call this function without a code change here